sys/include/uri_parser.h

```diff
@@ -67,15 +67,16 @@ typedef struct {
      */
     const char *zoneid;
 
-    const char *port;               /**< port */
+    const char *port_str;           /**< port as str */
+    uint16_t port;                  /**< port as uint16_t */
```

Comment:
For proper struct packing, it could make sense to move `port` down to the `*_len` members.